File: diploma/Menu/app/src/main/java/com/example/menu/MainActivity.java
package com.example.menu;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.Menu;
import android.view.MenuInflater;
import android.view.MenuItem;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
    }
    public boolean onCreateOptionsMenu(Menu menu) {
        MenuInflater inflater = getMenuInflater();
        inflater.inflate(R.menu.menu_example, menu);
        return true;
    }
    public boolean onOptionsItemSelected(MenuItem item) {
        switch (item.getItemId()) {
            case R.id.about_us:
                Toast.makeText(getApplicationContext(),"About Me", Toast.LENGTH_SHORT).show();
                return true;
            case R.id.contact_us:
                Toast.makeText(getApplicationContext(),"Contact Me", Toast.LENGTH_SHORT).show();
                return true;
            default:
                return super.onOptionsItemSelected(item);
        }
    }
}

File: DiceRoller/app/src/main/java/jay/nakum/diceroller/MainActivity.kt
package jay.nakum.diceroller

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Button
import android.widget.ImageView
import android.widget.TextView
import android.widget.Toast

/**
 * This activity allows the user to roll a dice and view the result
 * on the screen.
 */
class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        /** Do a dice roll when the app starts */
        rollDice()

        val rollButton: Button = findViewById(R.id.button)
        rollButton.setOnClickListener {
            /** Roll the dice when button is clicked */
            rollDice()
            Toast.makeText(this, "Dice Rolled!", Toast.LENGTH_SHORT).show()
        }
    }

    /**
     * Roll the dice and update the screen with the result.
     */
    private fun rollDice() {
        /** Create new Dice object with 6 sides */
        val dice = Dice(6)
        /** Roll the dice */
        val roll = dice.roll()

        /** Update the screen after the roll */
        val diceImage: ImageView = findViewById(R.id.imageView)
        val drawableResource = when(roll) {
            1 -> R.drawable.dice_1
            2 -> R.drawable.dice_2
            3 -> R.drawable.dice_3
            4 -> R.drawable.dice_4
            5 -> R.drawable.dice_5
            else -> R.drawable.dice_6
        }
        diceImage.setImageResource(drawableResource)
        diceImage.contentDescription = roll.toString()
        // val textView: TextView = findViewById(R.id.textView)
        // textView.text = roll
    }
}